### PR TITLE
Country select feature added with search

### PR DESCRIPTION
**Pull request by ShubhamVora-1206**
- Added country select dropdown in QueryBox component
- Updated search query API call to include country parameter
- Added Country enum to display selected countries in constants.js

image
image

### CODE REVIEW

**Comment by abhinandanmishra1** [comment]
@ShubhamVora-1206 format all the files and resolved conflicts

**Comment by ShubhamVora-1206** [comment]
- Refurbished product toggle Feature added
image

**Comment by ShubhamVora-1206** [comment]
Conflicts resolved

File: client/src/apis/recommendations.js [hunk 1]
```diff
@@ -4,7 +4,9 @@ export const getRecommendations = async (query) => {
   const axios = getAxios();
   return axios.get(`/search`, {
     params: {
-      query
+      query:query?.query,
```
Comment (abhinandanmishra1) [inline client/src/apis/recommendations.js:7]: can we destructure in the function parameter? `const getRecommendations = async ({ query, country, refurbished }) { ... }`

File: client/src/blocks/QueryBox/index.jsx [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
-
+import ReactFlagsSelect from "react-flags-select";
 import { IoSendSharp } from "react-icons/io5";
 
 export const QueryBox = ({ onSearch }) => {
   const ref = useRef();
   const [query, setQuery] = useState("");
+  const [country, setCountry] = useState("IN");
+  const [toggle, setToggle] = useState(false);
+  console.log(country);
```
Comment (abhinandanmishra1) [inline client/src/blocks/QueryBox/index.jsx:11]: please remove consoles

File: client/src/components/Chats/ChatNew/index.jsx [hunk 3]
```diff
@@ -10,16 +10,14 @@ import React, { useState } from "react";
 import { FeaturesDescription } from "./FeaturesDescription";
 import { useGetRecommendations } from "@app/hooks";
 
-export const ChatNew = ({showQuery}) => {
-  const [query, setQuery] = useState("");
-  console.log(query)
+export const ChatNew = ({ showQuery }) => {
+  const [query, setQuery] = useState({ query: "", country: "IN", refurbished:false });
+  console.log("query:",query);
```
Comment (abhinandanmishra1) [inline client/src/components/Chats/ChatNew/index.jsx:15]: remove console.

File: client/src/hooks/recommendations.js [hunk 4]
```diff
@@ -14,6 +14,6 @@ export const useGetRecommendations = (query) => {
             console.log(data)
             navigate(`/chats/${data.data.chatId}`)
         },
-        enabled: !!query
+        enabled: !!query.query
```
Comment (abhinandanmishra1) [inline client/src/hooks/recommendations.js:17]: instead of writing `query.query`, you can rename parameterName as `data` and write `data.query`

**Review by abhinandanmishra1 — COMMENTED** [review]
Please resolve the comments, after that code is good to merge

File: client/src/blocks/QueryBox/index.jsx [hunk 5]
```diff
@@ -1,13 +1,18 @@
+import { COUNTRY_MAP, CUSTOM_COUNTRY_LABELS } from "@app/utils/constants";
```
[the remark this answers is not in the document]
Comment (ShubhamVora-1206) [reply client/src/blocks/QueryBox/index.jsx:1]: I've kept it separate to differentiate utility functions and utility constants. When this code is shared, the reader would be able to track where the constants are and where the functions are. Let me know if this makes sense

File: client/src/apis/recommendations.js [hunk 6]
```diff
@@ -4,7 +4,8 @@ export const getRecommendations = async (query) => {
...
+      country:query?.country
```
[the remark this answers is not in the document]
Comment (ShubhamVora-1206) [reply client/src/apis/recommendations.js]: resolved

**Pull request merged**